### PR TITLE
Refactoring with single message processor , single pool configuration…

### PR DESCRIPTION
**Pull request by Yasara123**
1) Chane architecture to keep single message processor and single bootstrap pool configuration
2) Remove redundant classes
3) Add more descriptions
4) add osgi test for sink
5) formating and renaming 
6) move default client-store and truststore jks files from conf/security to resources/security 
7) remove bootstrap boss and worker thread pool size parameter from source parameters and add it as system parameter

### CODE REVIEW

File: component/src/test/java/org/wso2/extension/siddhi/io/http/source/HttpBasicTests.java [hunk 1]
```diff
@@ -127,116 +139,125 @@ public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
...
+            Assert.assertEquals(1, executionPlanRuntime.getSources().size());
         } catch (Exception e) {
-            Assert.assertEquals("Listener URL http://localhost:9005/endpoints/" +
-                    "RecPro already connected in StreamDefinition{id='inputStreamB', attributeList" +
-                    "=[Attribute{id='name', type=STRING}, Attribute{id='age', type=INT}, Attribute{id='country', " +
-                    "type=STRING}], annotations=[Annotation{name='source', elements=[Element{key='type', value='http'}"
-                    + ", Element{key='receiver.url', value='http://localhost:9005/endpoints/RecPro'}, Element{key=" +
-                    "'basic.auth.enabled', value='false'}], annotations=[Annotation{name='map', " +
-                    "elements=[Element{key='type', value='xml'}], annotations=[]}]}]}", e.getMessage());
+            throw new RuntimeException(e);
```
Comment (tishan89) [inline component/src/test/java/org/wso2/extension/siddhi/io/http/source/HttpBasicTests.java]: What is the reason for doing this?

File: component/src/test/java/org/wso2/extension/siddhi/io/http/source/HttpsSourceTestCaseForSSL.java [hunk 2]
```diff
@@ -46,217 +46,181 @@
...
-        inMemoryConfigManager.generateConfigReader("source", "http");
-        siddhiManager.setConfigManager(inMemoryConfigManager);
-        siddhiManager.setExtension("xml-input-mapper", XmlSourceMapper.class);
-        String inStreamDefinition = "" + "@source(type='http', @map(type='xml'), "
-                + "receiver.url='https://localhost:9095/endpoints/RecPro', " + "basic.auth.enabled='false'" + ")"
-                + "define stream inputStream (name string, age int, country string);";
-        String query = ("@info(name = 'query1') " + "from inputStream " + "select *  " + "insert into outputStream;");
-        ExecutionPlanRuntime executionPlanRuntime = siddhiManager
-                .createExecutionPlanRuntime(inStreamDefinition + query);
+//    /**
```
Comment (tishan89) [inline component/src/test/java/org/wso2/extension/siddhi/io/http/source/HttpsSourceTestCaseForSSL.java]: Let's remove commented out code

File: component/src/test/java/org/wso2/extension/siddhi/io/http/source/HttpsSourceTestCaseForSSL.java [hunk 3]
```diff
@@ -268,36 +232,108 @@ public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
...
+        org.hamcrest.MatcherAssert.assertThat(log.get(4).getLevel(), is(Level.ERROR));
+        org.hamcrest.MatcherAssert.assertThat(log.get(3).getLoggerName(), is("org.wso2.extension.siddhi.io" +
+                ".http.source.util.HttpTestUtil"));
         Thread.sleep(100);
         Assert.assertEquals(receivedEventNameList.toString(), expected.toString());
         executionPlanRuntime.shutdown();
     }
-    class TestAppender extends AppenderSkeleton {
+
+//    /**
```
Comment (tishan89) [inline component/src/test/java/org/wso2/extension/siddhi/io/http/source/HttpsSourceTestCaseForSSL.java]: Let's remove commented out code

File: tests/osgi-tests/src/test/java/org/wso2/extension/siddhi/io/http/test/osgi/HttpSink.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (tishan89) [inline tests/osgi-tests/src/test/java/org/wso2/extension/siddhi/io/http/test/osgi/HttpSink.java]: Wrong license year

File: tests/osgi-tests/src/test/java/org/wso2/extension/siddhi/io/http/test/osgi/HttpSink.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                        .artifactId("org.wso2.extension.io.http.test.distribution")
+                        .type("zip")
+                        .versionAsInProject()),
+                copyOSGiLibBundle(maven()
+                        .artifactId("siddhi-io-http")
+                        .groupId("org.wso2.extension.siddhi.io.http")
+                        .versionAsInProject()),
+                systemProperty("java.security.auth.login.config")
+                        .value(Paths.get("conf", "security", "carbon-jaas.config").toString())
+                //CarbonDistributionOption.debug(5005)
```
Comment (tishan89) [inline tests/osgi-tests/src/test/java/org/wso2/extension/siddhi/io/http/test/osgi/HttpSink.java]: Remove commented code

File: tests/osgi-tests/src/test/java/org/wso2/extension/siddhi/io/http/test/osgi/sink/util/HttpServerListener.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+/**
+ * Test Server Listener Manger.
+ */
+public class HttpServerListener implements HttpHandler {
+    private AtomicBoolean isEventArraved = new AtomicBoolean(false);
+    private StringBuilder strBld;
+    private Headers headers;
+    private static final Logger logger = Logger.getLogger(HttpServerListener.class);
+
+    HttpServerListener() {
```
Comment (tishan89) [inline tests/osgi-tests/src/test/java/org/wso2/extension/siddhi/io/http/test/osgi/sink/util/HttpServerListener.java]: Are we intentionally keeping protected access modifier? Still it is best practice to specify the access modifier.

File: tests/osgi-tests/src/test/java/org/wso2/extension/siddhi/io/http/test/osgi/sink/util/HttpServerListenerHandler.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+import org.apache.log4j.Logger;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+
+/**
+ *Http test sever listener.
+ */
+public class HttpServerListenerHandler implements Runnable {
+    public HttpServerListener getServerListener() {
```
Comment (tishan89) [inline tests/osgi-tests/src/test/java/org/wso2/extension/siddhi/io/http/test/osgi/sink/util/HttpServerListenerHandler.java]: Let's add the method below variables

File: component/src/test/java/org/wso2/extension/siddhi/io/http/source/HttpBasicTests.java (hunk 1 above)
Comment (Yasara123) [reply component/src/test/java/org/wso2/extension/siddhi/io/http/source/HttpBasicTests.java]: Fixed in 602e96bfa91ea786afe9f9bd414b1bc8b1787631

File: component/src/test/java/org/wso2/extension/siddhi/io/http/source/HttpsSourceTestCaseForSSL.java (hunk 2 above)
Comment (Yasara123) [reply component/src/test/java/org/wso2/extension/siddhi/io/http/source/HttpsSourceTestCaseForSSL.java]: Fixed in 602e96bfa91ea786afe9f9bd414b1bc8b1787631

File: component/src/test/java/org/wso2/extension/siddhi/io/http/source/HttpsSourceTestCaseForSSL.java (hunk 3 above)
Comment (Yasara123) [reply component/src/test/java/org/wso2/extension/siddhi/io/http/source/HttpsSourceTestCaseForSSL.java]: Fixed in 602e96bfa91ea786afe9f9bd414b1bc8b1787631

File: tests/osgi-tests/src/test/java/org/wso2/extension/siddhi/io/http/test/osgi/HttpSink.java (hunk 4 above)
Comment (Yasara123) [reply tests/osgi-tests/src/test/java/org/wso2/extension/siddhi/io/http/test/osgi/HttpSink.java]: Fixed in 4970a6d52fe797b43d60461fa05a71b8e80fd262

File: tests/osgi-tests/src/test/java/org/wso2/extension/siddhi/io/http/test/osgi/HttpSink.java (hunk 5 above)
Comment (Yasara123) [reply tests/osgi-tests/src/test/java/org/wso2/extension/siddhi/io/http/test/osgi/HttpSink.java]: Fixed in 156b93ab839dfc225e0a915f8951b61db7afa2ab

File: tests/osgi-tests/src/test/java/org/wso2/extension/siddhi/io/http/test/osgi/sink/util/HttpServerListenerHandler.java (hunk 7 above)
Comment (Yasara123) [reply tests/osgi-tests/src/test/java/org/wso2/extension/siddhi/io/http/test/osgi/sink/util/HttpServerListenerHandler.java]: Fixed 156b93ab839dfc225e0a915f8951b61db7afa2ab

File: tests/osgi-tests/src/test/java/org/wso2/extension/siddhi/io/http/test/osgi/sink/util/HttpServerListener.java (hunk 6 above)
Comment (Yasara123) [reply tests/osgi-tests/src/test/java/org/wso2/extension/siddhi/io/http/test/osgi/sink/util/HttpServerListener.java]: Fixed in Fixed in 6bf095af9d7a4c00220069badb50517cec553139

**Pull request merged**